### PR TITLE
Set application name for API calls

### LINKED ISSUE #41: Set application name when calling Google APIs
```
Feb 01, 2017 12:14:28 PM com.google.api.client.googleapis.services.AbstractGoogleClient <init>
WARNING: Application name is not set. Call Builder#setApplicationName.
```

The only API being used at the moment is OAuth2 API though.

A default name could be "google-ide-login", but we should allow clients of this library to set a custom application name.

### PR DESCRIPTION
**Pull request by chanseokoh**
Fixes #41.

### CODE REVIEW

File: src/main/java/com/google/cloud/tools/ide/login/GoogleLoginState.java [hunk 1]
```diff
@@ -124,6 +128,10 @@ public GoogleLoginState(String clientId, String clientSecret, Set<String> oAuthS
     retrieveSavedCredentials();
   }
 
+  public void setApiApplicationName(String apiApplicationName) {
+    this.apiApplicationName = apiApplicationName;
```
Comment (elharo) [inline src/main/java/com/google/cloud/tools/ide/login/GoogleLoginState.java]: checkNotNull?

File: src/main/java/com/google/cloud/tools/ide/login/GoogleLoginState.java [hunk 2]
```diff
@@ -124,6 +128,10 @@ public GoogleLoginState(String clientId, String clientSecret, Set<String> oAuthS
     retrieveSavedCredentials();
   }
 
+  public void setApiApplicationName(String apiApplicationName) {
```
Comment (elharo) [inline src/main/java/com/google/cloud/tools/ide/login/GoogleLoginState.java]: Maybe just setApplicationName

**Comment by chanseokoh** [comment]
PTAL

**Comment by chanseokoh** [comment]
PATL @elharo

File: src/main/java/com/google/cloud/tools/login/GoogleLoginState.java [hunk 3]
```diff
@@ -58,6 +58,8 @@
 
   private static final String OAUTH2_NATIVE_CALLBACK_URL = GoogleOAuthConstants.OOB_REDIRECT_URI;
 
+  private static final String DEFAULT_APPLICATION_NAME = "google-ide-login";
```
Comment (elharo) [inline src/main/java/com/google/cloud/tools/login/GoogleLoginState.java]: This is only used in one place to set another field, right? If so there's no real need for this constant.

**Pull request merged**